### PR TITLE
feat: improve the UX for password protected documents

### PR DESCRIPTION
**Pull request by harkiratsm**
Current Behaviour - 
Presently, the behavior is quite default, which does not align with the theme. Additionally, the alert box appears to be somewhat stubborn.

https://github.com/documenso/documenso/assets/71957674/d681bef6-7ee3-4f27-b6db-34c6aaec6139

This PR fixes - 
I am overriding the default behavior by adding a password dialog component, which will only appear for password-protected documents.

https://github.com/documenso/documenso/assets/71957674/b1d05968-3c15-4abe-810f-e4186cfd1fcb

### CODE REVIEW

**Review by adithyaakrishna — APPROVED** [review]
@harkiratsm This would be a nice addition! Thank you for the contribution 🎉

**Review by catalinpit — CHANGES_REQUESTED** [review]
Nice work! But I get this error when the dialog box opens. Could you have a look and fix this, please?

CleanShot 2023-12-29 at 11 05 01

File: packages/ui/primitives/pdf-viewer.tsx [hunk 1]
```diff
@@ -77,6 +82,14 @@ export const PDFViewer = ({
     setNumPages(doc.numPages);
     onDocumentLoad?.(doc);
   };
+  
+  const handlePasswordSubmit = () => {
```
Comment (dephraiim) [inline packages/ui/primitives/pdf-viewer.tsx:86]: Can you rename this to `onPasswordSubmit` to match with the convention?

File: packages/ui/primitives/pdf-viewer.tsx [hunk 2]
```diff
@@ -169,11 +182,26 @@ export const PDFViewer = ({
           <PDFLoader />
         </div>
       ) : (
+        <>
```
Comment (dephraiim) [inline packages/ui/primitives/pdf-viewer.tsx:185]: Can you format the React Fragment well?

File: packages/ui/primitives/document-password-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  isError?: boolean;
+}
+
+export const PasswordDialog = ({ open, onOpenChange, handleSubmit, isError, setPassword }: PasswordDialogProps) => {
+  return (
+    <Dialog open={open} onOpenChange={onOpenChange}>
+      <DialogContent className="max-w-md">
+        <DialogHeader>
+          <DialogTitle>Password Required</DialogTitle>
+          <DialogDescription>
```
Comment (dephraiim) [inline packages/ui/primitives/document-password-dialog.tsx:29]: Can you maintain `This document is password protected. Please enter the password to view the document.` as the description and show the error below the input?

Much like what we do on the sign up page

**Review by dephraiim — CHANGES_REQUESTED** [review]

File: packages/ui/primitives/document-password-dialog.tsx (hunk 3 above)
Comment (harkiratsm) [reply packages/ui/primitives/document-password-dialog.tsx]: image


Updated !

**Comment by Mythie** [comment]
This looks really cool! Would you be willing to attempt the storing of the password entered as part of the documents meta so we can unlock the document during signing as well?

If not we're happy to have a team member work on this 🙌

**Comment by harkiratsm** [comment]
Will do! Thanks! 🚀

**Comment by harkiratsm** [comment]
User Tests - 

- [ ] Creation of a protected document
- [ ]  Signing a protected document
- [ ]  Templating a protected document
- [ ]  Editing a protected document


The UX is enhanced when working with password-protected documents. Currently, the password is stored in the documentMeta, facilitating the automatic opening of the PDF viewer for users during both signing and editing processes.

**Review by Mythie — APPROVED** [review]

**Comment by Mythie** [comment]
Awesome work Harkirat! 🎉 

I've made a few changes which you can hear about here: https://www.youtube.com/watch?v=89j3Ukgi_go

Thanks once again for the awesome work 💪

**Comment by harkiratsm** [comment]
Thank you for your improvements. @Mythie

**Pull request merged**